### PR TITLE
Fixes Micro-Bomb implant dropping your stuff when it shouldn't

### PR DESCRIPTION
**Pull request by cowbot92**
# Document the changes in your pull request

microbomb implants dont give everyone everything you own

# Testing

wouldnt make it if it didnt test it

# Changelog

:cl:  

bugfix: fixes microbomb implants not deleting ur stuff

/:cl:

### CODE REVIEW

File: code/modules/mob/living/death.dm [hunk 1]
```diff
@@ -19,6 +19,23 @@ GLOBAL_VAR_INIT(permadeath, FALSE)
 	SEND_SIGNAL(src, COMSIG_LIVING_GIBBED, no_brain, no_organs, no_bodyparts)
 	qdel(src)
 
+/mob/living/proc/gib_no_drop(no_brain, no_organs, no_bodyparts)
```
Comment (iamthesilly) [inline code/modules/mob/living/death.dm]: Add an argument for not dropping items to `gib()` instead of adding duplicate code

**Review by SapphicOverload — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/death.dm (hunk 1 above)
Comment (cowbot92) [reply code/modules/mob/living/death.dm]: but I cant because of the proc
image

Comment (cowbot92) [reply code/modules/mob/living/death.dm]: nvm I see what you mean, done

File: code/modules/mob/living/death.dm [hunk 2]
```diff
@@ -1,22 +1,20 @@
 GLOBAL_VAR_INIT(permadeath, FALSE)
 
-/mob/living/proc/gib(no_brain, no_organs, no_bodyparts)
+/mob/living/proc/gib(no_brain, no_organs, no_bodyparts, unequip_everything)
```
Comment (iamthesilly) [inline code/modules/mob/living/death.dm]: This should still unequip by default, also change the microbomb's `gib(1)` to `gib(no_brain = TRUE, unequip_everything = FALSE)`
```suggestion
/mob/living/proc/gib(no_brain, no_organs, no_bodyparts, unequip_everything = TRUE)
```

File: code/modules/mob/living/death.dm [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
 	var/prev_lying = lying
 	if(stat != DEAD)
 		death(TRUE)
 
-	unequip_everything()
```
Comment (iamthesilly) [inline code/modules/mob/living/death.dm:8]: You still need to unequip things, just check `if(unequip_everything)` first.

File: code/modules/mob/living/death.dm [hunk 4]
```diff
@@ -1,22 +1,20 @@
...
-
 	if(!prev_lying)
 		gib_animation()
 
-	spill_organs(no_brain, no_organs, no_bodyparts)
+	spill_organs(no_brain, no_organs, no_bodyparts, unequip_everything)
```
Comment (iamthesilly) [inline code/modules/mob/living/death.dm]: `spill_organs` has no argument for this and doesn't unequip things anyway.
```suggestion
	spill_organs(no_brain, no_organs, no_bodyparts)
```

**Review by SapphicOverload — CHANGES_REQUESTED** [review]

**Comment by cowbot92** [comment]
im too stupid to actually fix this, someone else can do it

**Comment by cowbot92** [comment]
nevermind im actually too stupid to fix this @Moltijoe broke it he can fix it

**Pull request closed without merging**